### PR TITLE
Fix Offchain Community Creation

### PR DESCRIPTION
**Pull request by true-eye**
https://www.notion.so/hicommonwealth/Fix-Offchain-Community-Creation-c37e10b58d1c46adb60c52a05c62ae81

- Show the `+` button to all users instead of only site admin users
- Display `Default Chain` dropdown on Offchain Community Create Modal to select one of already joined chains.
- Fix `/createCommunity` api. (only ethereum user could create. now it is enabled for all)

### CODE REVIEW

**Comment by jnaviask** [comment]
Some notes from QA:
- I can view the + button and modal even when I'm not logged in. It gives me a 401 unauthorized error in the console while showing no feedback on the modal if I try to submit.
- Anyone can view the Substrate tab even though it's admin-only to create (and it gives an API error...).
- Not sure what's up with this list of base chains:
image

- Website/Discord/Telegram/etc fields don't seem to persist after creating them -- I made a new community with some and they didn't show up in the Manage Community modal afterward. There also is probably no validation that someone doesn't just use phishing links for discord telegram etc.
- No visual feedback (no error text or popup) that my community creation failed because title was over 255 characters, or for other errors like duplicate title.
- I was, however, able to make a very long title and description which goes off the page:
2021-08-10-141915_1800x376_scrot
2021-08-10-142210_1089x355_scrot

This is all important stuff to prevent abuse if we're going to go public with this feature, at least IMO. Curious what @dillchen thinks.

**Comment by true-eye** [comment]
Cool, I will fix the above

File: server/routes/createCommunity.ts [hunk 1]
```diff
@@ -33,6 +42,20 @@ const createCommunity = async (models, req: Request, res: Response, next: NextFu
   if (req.body.invitesEnabled !== 'true' && req.body.invitesEnabled !== 'false') {
     return next(new Error(Errors.NoInvitesEnableldSetting));
   }
+
+  const { website, discord, element, telegram, github } = req.body;
+  if (website && !urlHasValidHTTPPrefix(website)) {
+    return next(new Error(Errors.InvalidWebsite));
+  } else if (discord && !urlHasValidHTTPPrefix(discord)) {
```
Comment (jnaviask) [inline server/routes/createCommunity.ts:49]: This isn't really sufficient checking for whether a link actually goes to discord/github/element etc, but it's fine for now.

**Review by jnaviask — COMMENTED** [review]
Can we also make it disable the buttons while we're waiting for a server response, so the user doesn't send the request multiple times?

**Comment by true-eye** [comment]
> Can we also make it disable the buttons while we're waiting for a server response, so the user doesn't send the request multiple times?

yeah. I will do

File: server/routes/createCommunity.ts (hunk 1 above)
Comment (true-eye) [reply server/routes/createCommunity.ts:49]: yes. exactly. just followed the other parts of the code

**Comment by true-eye** [comment]
I fixed the saving state. Please review

File: client/scripts/views/modals/create_community_modal.ts [hunk 2]
```diff
@@ -165,6 +170,8 @@ const OffchainCommunityForm: m.Component<OffchainCommunityFormAttrs, OffchainCom
             m.route.set(`/${res.result.id}`);
           }).catch((err: any) => {
             notifyError(err.responseJSON?.error || 'Creating new community failed');
```
[the remark this answers is not in the document]
Comment (true-eye) [reply client/scripts/views/modals/create_community_modal.ts:172]: I thought `done` is called on both `then` & `catch`?

[the remark this answers is not in the document]
Comment (jnaviask) [reply client/scripts/views/modals/create_community_modal.ts:172]: Ah you're right, my bad. Will do a quick QA now then should be good to go.

[the remark this answers is not in the document]
Comment (true-eye) [reply client/scripts/views/modals/create_community_modal.ts:172]: @jnaviask you are right, I've fixed with `always`

**Comment by jnaviask** [comment]
Much better -- only thing weird is that list of default chains. I don't understand why it's populated the way it is.

**Comment by true-eye** [comment]
> Much better -- only thing weird is that list of default chains. I don't understand why it's populated the way it is.

I think, it is supposed to display the chains which the current user joined. @dillchen please confirm

**Pull request merged**